modiscolite/meme_writer.py

```diff
@@ -0,0 +1,158 @@
+# Create a class that can be used to write MEME files.
+# Authors: Ivy Raine <ivy.ember.raine@gmail.com>, Jacob Schreiber <jmschreiber91@gmail.com>
+from typing import List, Optional
+import numpy as np
+
+from os import PathLike
+
+class MEMEWriterMotif:
+	"""Class for handling Motif for MEME file writing."""
+
+	def __init__(
+		self,
+		name: str,
+		probability_matrix: np.ndarray,
+		# `alphabet_length` is known as 'alength' in MEME Suite.
+		alphabet_length: int,
```

Comment:
Are there cases where this isn't simply derived from the probability matrix shape? If not, maybe we shouldn't include it? It's just offering the user a degree of freedom that can lead to errors.

Reply:
Judging by the spec, they allow the user to provide an alength separate from the derived version:
> The "alength= alphabet length" and "w= motif length" can be derived from the matrix if they are not specified, provided there is an empty line following the letter probability matrix.
But yeah, I'll make it optional. I won't add support for custom alphabets here though- it's beyond the scope of the project. More of an issue for if we move the meme writer to a separate repo.

Reply:
It's weird to me that they bother accepting the dimensions of the matrix they're about to read. I guess it's to know how many lines to read?  Agree wrt custom alphabets.